src_c/surface.c

```diff
@@ -1907,23 +1907,34 @@ surf_blits(pgSurfaceObject *self, PyObject *args, PyObject *keywds)
     PyObject *special_flags = NULL;
     PyObject *ret = NULL;
     PyObject *retrect = NULL;
-    Py_ssize_t itemlength;
+    Py_ssize_t itemlength, sequencelength, curriter = 0;
     int doreturn = 1;
     int bliterrornum = 0;
+    int issequence = 0;
     static char *kwids[] = {"blit_sequence", "doreturn", NULL};
     if (!PyArg_ParseTupleAndKeywords(args, keywds, "O|i", kwids, &blitsequence,
                                      &doreturn))
         return NULL;
 
-    if (doreturn) {
-        ret = PyList_New(0);
-        if (!ret)
-            return NULL;
-    }
-    if (!PyIter_Check(blitsequence) && !PySequence_Check(blitsequence)) {
+    if (!PyIter_Check(blitsequence) &&
+        !(issequence = PySequence_Check(blitsequence))) {
         bliterrornum = BLITS_ERR_SEQUENCE_REQUIRED;
         goto bliterror;
     }
+
+    if (doreturn) {
+        if (issequence && (sequencelength = PySequence_Size(blitsequence))) {
```

Comment:
`PySequence_Size` returns -1 on a failure, and this should be handled here. The code would be more clearer if the `sequencelength = PySequence_Size(blitsequence)` part was inside the conditional block

Reply:
resolved it